mautic/mautic
# Move console script to bin folder. Fix Transifex issues.

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8322
| BC breaks? | `app/console` is no longer available. Use `bin/console` instead.
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
`app/console` script is still included into the distribution package of Mautic 3.
Also, there are Transifex related errors in the console when we try to run console commands.
E.g.
```
[WARNING] Some commands could not be registered:
In getTransifex_FactoryService.php line 8:                                                              
  Class 'RicardoFiorani\GuzzlePsr18Adapter\Client' not found                                                               

In getTransifex_FactoryService.php line 8:                                                              
  Class 'RicardoFiorani\GuzzlePsr18Adapter\Client' not found  
```

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Download and install Mautic  `3.0.0-alpha` from https://github.com/mautic/mautic/releases/tag/3.0.0-alpha.
2. Try to run a console command. You will get Transfex related errors in the console (see above).

#### Steps to test this PR:
1. Install Mautic from this branch: https://github.com/anton-vlasenko/mautic/commits/move-console-script-to-bin-folder (development version).
2. Add a `3.0.0-alpha` tag to the latest commit in that branch (don't push it! to the repository; this is needed for testing only. You have too add it using --force flag).
```bash
git tag -a 3.0.0-alpha <commit_hash> -m "3.0.0-alpha" --force
```
3. Make sure your diff.renamelimit value is high enough.
```bash
git config diff.renamelimit 1000000
```
4. Run `php build/package_release.php`.
5. Install Mautic using the package that you built on the previous step (3.0.0-alpha.zip).
6. Check that there is no app/console file in your new Mautic installation.
7. Try to run console commands. There must be no Transifex errors.

#### List backwards compatibility breaks:
1. `app/console` is no longer available. Use `bin/console` instead.
